### PR TITLE
Migrate Package Specific Dependencies from Image to package.xml

### LINKED ISSUE #242: Explore ROS package dependencies
### Purpose

Clean up dependencies

### Changes

Is it possible to specify the pip packages required in the ROS package and have those be installed when running the setup task?

### Resources

- https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html#customize-package-xml

### PR DESCRIPTION
**Pull request by SPDonaghy**
### Description

- Partially Resolves #242 

Some Python dependencies needed for the local pathfinding ROS package were moved from `base-dev.Dockerfile` to `package.xml`  in the local_pathfinding repository

### Verification

- After removing some dependencies from `base-dev.Dockerfile`, I ran the build images workflow and set the new image tag in the main Dockerfile.
- I then rebuilt the container and verified that the pip3 dependencies in `base-dev.Dockerfile` were installed in the workspace but the dependencies that were moved to `package.xml` were not.
- Next, I ran the setup task and then verified the remaining dependencies were installed.

### Resources

- This PR only dealt with local_pathfinding dependencies hence only partially resolving the mentioned Sailbot Workspace issue.
- [Here](https://github.com/UBCSailbot/local_pathfinding/pull/90#pullrequestreview-1898618098) is the related local_pathfinding PR

### To Do
- [x] [Patrick] Cleanup dockerfile
- [x] [Sean] revert local pathfinding branch to main once its PR is merged
- [ ] Migrate NET package dependencies to package.xml
    - [ ] "install base apt dependencies" section
    - [ ] "install some clang tools and googletest for network systems" section

### CODE REVIEW

**Comment by patrick-5546** [comment]
Setup is run in the tests run: https://github.com/UBCSailbot/sailbot_workspace/blob/b602a2023b7d0c83c96026ba50fb457be09f6c1f/.github/actions/test/run.sh#L5

I think it is failing because it is pulling the main branch of local pathfinding. You can test out your branch by changing this line: https://github.com/UBCSailbot/sailbot_workspace/blob/b602a2023b7d0c83c96026ba50fb457be09f6c1f/src/polaris.repos#L48

File: .devcontainer/base-dev/base-dev.Dockerfile [hunk 1]
```diff
@@ -377,10 +369,13 @@ ENV DEBIAN_FRONTEND=
...
 RUN pip3 install \
-    # to be able to run juypter notebooks
+    # for juypter notebooks
     ipykernel \
     # for integration_tests package
-    types-PyYAML
+    types-PyYAML \
+    # for local pathfinding
+    plotly \
+    flask
```
Comment (patrick-5546) [inline .devcontainer/base-dev/base-dev.Dockerfile:378]: Should pip install these packages in the notebook itself, for example: https://github.com/UBCSailbot/notebooks/blob/main/local_pathfinding/coord_systems.ipynb

File: .devcontainer/base-dev/base-dev.Dockerfile [hunk 2]
```diff
@@ -377,10 +369,13 @@ ENV DEBIAN_FRONTEND=
 
 # install dev python3 dependencies
 RUN pip3 install \
-    # to be able to run juypter notebooks
+    # for juypter notebooks
     ipykernel \
     # for integration_tests package
-    types-PyYAML
+    types-PyYAML \
```
Comment (patrick-5546) [inline .devcontainer/base-dev/base-dev.Dockerfile:375]: Can migrate this to its package.xml: https://github.com/UBCSailbot/sailbot_workspace/blob/user/SPDonaghy/migrate-package-specific-dependencies-from-image-to-packages-242/src/integration_tests/package.xml

File: .devcontainer/base-dev/base-dev.Dockerfile (hunk 1 above)
Comment (SPDonaghy) [reply .devcontainer/base-dev/base-dev.Dockerfile:378]: Which notebook do you mean?

**Comment by SPDonaghy** [comment]
> I think it is failing because it is pulling the main branch of local pathfinding. You can test out your branch by changing this line:

Hm I tried this but its still not working.

This is what you meant right?
image

Running the setup task manually and then the test task works.
But just running test does not. It also doesn't complain if I put a wrong branch name in polaris.repos either.

**Comment by patrick-5546** [comment]
> Hm I tried this but its still not working.

It timed out, which is a known bug: #231

File: .devcontainer/base-dev/base-dev.Dockerfile (hunk 1 above)
Comment (patrick-5546) [reply .devcontainer/base-dev/base-dev.Dockerfile]: nvm these are the development-only dependencies of local pathfinding. See my commit in your local pathfinding pr

**Review by patrick-5546 — APPROVED** [review]
Can merge your local pathfinding pr in, change the branch back to main here, then merge this pr in

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
Actually let's keep this open for now and use it to clean up our dependencies across the software team. I'll add a to do list to this PR

**Comment by patrick-5546** [comment]
> I'll make separate PRs in each repo to update the package.xml files one by one

As far as I can tell what's left is syncing with NET/diagnostics about what dependencies can be migrated to their repo

**Comment by SPDonaghy** [comment]
> After rerunning it passes

So on this branch, when you rebuild and then run the test task, it doesn't error out for you? I'm getting module not found for scipy from controller and pyproj from local_path

**Review by hhenry01 — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/ubcsailbot/sailbot_workspace/dev:fix-building-pre-base
+FROM ghcr.io/ubcsailbot/sailbot_workspace/dev:combined-pip3-installs3
```
Comment (patrick-5546) [inline .devcontainer/Dockerfile:1]: Have you updated this to the latest tag? If so, good to merge

**Review by patrick-5546 — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (SPDonaghy) [reply .devcontainer/Dockerfile:1]: Will update now

**Pull request merged**